src/Compilers/Core/Rebuild/BuildConstructor.cs

```diff
@@ -103,7 +103,8 @@ public BuildConstructor(ILogger logger)
             var pdbCompilationOptions = optionsReader.GetMetadataCompilationOptions();
 
             var langVersionString = pdbCompilationOptions.GetUniqueOption("language-version");
-            var optimization = pdbCompilationOptions.GetUniqueOption("optimization");
+            pdbCompilationOptions.TryGetUniqueOption("optimization", out var optimization);
+            pdbCompilationOptions.TryGetUniqueOption("platform", out var platform);
```

Comment:
```suggestion
            pdbCompilationOptions.TryGetUniqueOption(CompilationOptionNames.Optimization, out var optimization);
            pdbCompilationOptions.TryGetUniqueOption(CompilationOptionNames.Platform, out var platform);
```

Reply:
This is fine, but we should do the rest too.